### PR TITLE
[WIP] For #9480 feat(project): Create v6 endpoint to get fml errors

### PR DESCRIPTION
**Pull request by eliserichards**
Because

- We want to create a rest endpoint that will [fetch the FML errors](https://mozilla-hub.atlassian.net/browse/EXP-3871) from the `NimbusFmlLoader

This commit

- Creates the exoskeleton of the api --
   - Create a `POST` request 
   - Define the schema: the query will contain `application`, `channel`, and the blob that the FML will use to find errors
   - Return errors and a valid `Response`:
      - The FML loader will be hooked up in EXP-3871, so the return value is just a placeholder for this PR
      - The parsing of the errors returned from the FML will also be handled in EXP-3871 on the frontend side, not in python

### CODE REVIEW

**Comment by jaredlockhart** [comment]
If this is being consumed by nimbus ui, shouldn't it go in V5?

**Comment by eliserichards** [comment]
> If this is being consumed by nimbus ui, shouldn't it go in V5?

@jaredlockhart this is a rest endpoint that has nothing to do with graphql/models, so I believe it should live in the v6 api 🤔

**Comment by eliserichards** [comment]
hmm ... integration tests failing 🤔 
```
==================================== ERRORS ====================================
______________ ERROR collecting nimbus/test_desktop_targeting.py _______________
.tox/integration-test-nimbus/lib/python3.8/site-packages/requests/models.py:971: in json
    return complexjson.loads(self.text, **kwargs)
/usr/lib/python3.8/json/__init__.py:357: in loads
    return _default_decoder.decode(s)
/usr/lib/python3.8/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
/usr/lib/python3.8/json/decoder.py:355: in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
E   json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:
nimbus/test_desktop_targeting.py:9: in <module>
    @pytest.fixture(params=helpers.load_targeting_configs())
nimbus/utils/helpers.py:139: in load_targeting_configs
    config_data = load_config_data()
nimbus/utils/helpers.py:28: in load_config_data
    return load_graphql_data(
nimbus/utils/helpers.py:14: in load_graphql_data
    return requests.post(
.tox/integration-test-nimbus/lib/python3.8/site-packages/requests/models.py:975: in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
E   requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
...
=========================== short test summary info ============================
ERROR nimbus/test_desktop_targeting.py - requests.exceptions.JSONDecodeError:...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
================= 1 skipped, 122 deselected, 1 error in 10.17s =================
ERROR: InvocationError for command /code/experimenter/tests/integration/.tox/integration-test-nimbus/bin/pytest --verify-base-url --base-url https://nginx/nimbus/ --html=test-reports/report.htm --self-contained-html --reruns-delay 30 --driver Firefox nimbus/ -k FIREFOX_DESKTOP -m desktop_enrollment --reruns 1 -vvv (exited with code 2)
___________________________________ summary ____________________________________
ERROR:   integration-test-nimbus: commands failed
make: *** [Makefile:226: integration_test_nimbus] Error 1

Exited with code exit status 2
```

File: experimenter/experimenter/experiments/api/v6/urls.py [hunk 1]
```diff
@@ -18,4 +20,12 @@
...
 )
+
 urlpatterns = router.urls
+urlpatterns += [
+    re_path(
+        r"nimbus-fml-diagnostics",
+        NimbusFmlDiagnosticsApi.as_view(),
+        name="nimbus-fml-diagnostics-api",
+    ),
+]
```
Comment (jhugman) [inline experimenter/experimenter/experiments/api/v6/urls.py:31]: Should this use the `router.register` more directly?

```suggestion
router.register(
    r"nimbus-fml-diagnostics",
    NimbusFmlDiagnosticsApi.as_view(),
    name="nimbus-fml-diagnostics-api",
)
```

File: experimenter/experimenter/experiments/api/v6/views.py [hunk 2]
```diff
@@ -37,3 +48,55 @@ class NimbusExperimentFirstRunViewSet(NimbusExperimentViewSet):
...
+        ]
+        return operation
+
+
+class NimbusFmlDiagnosticsApi(APIView):
+    def post(self, request):  # pragma: no cover
+        self.schema = NimbusFmlDiagnosticsApiSchema
+        application = request.POST.get("application")
+        channel = request.POST.get("channel")
+        request.POST.get("blob")
```
Comment (jhugman) [inline experimenter/experimenter/experiments/api/v6/views.py:84]: Will this be eventually:
```suggestion
        blob = request.POST.get("blob")
```

File: experimenter/experimenter/experiments/tests/api/v6/test_views.py [hunk 3]
```diff
@@ -179,3 +182,89 @@ def test_list_view_serializes_live_first_run_experiments(self):
...
+                                "default": "ic_fox",
+                            },
+                            "message-content": {
+                                "description": "Text content to show the user",
+                                "default": "msg_thankyou",
+                            },
+                        },
+                    }
+                }
+            }
```
Comment (jhugman) [inline experimenter/experimenter/experiments/tests/api/v6/test_views.py:248]: This look _like_ a sample FML in JSON. Is this necessarily complicated, or can you use a simpler string for this test case?

File: docs/experimenter/openapi-schema.json [hunk 4]
```diff
@@ -1613,6 +1613,39 @@
...
+              }
+            },
+            "description": ""
+          }
+        },
+        "tags": [
+          "Nimbus: Public"
+        ]
+      }
+    },
```
Comment (jhugman) [inline docs/experimenter/openapi-schema.json:1648]: I don't know if these docs files are just docs, or whether they're significant to the working of the code.

If they are the latter, then filing in the `schema` objects may be the missing piece to what we discussed yesterday. 

Either way, they endpoint should be documented before landing this PR.

**Review by jhugman — COMMENTED** [review]
Drive by review! Keep going!

**Comment by jaredlockhart** [comment]
@eliserichards No i mean, V5 contains all the read/write endpoints that are used by the frontend regardless of their structure.  V6 is the public read only API.  I don't think this is read only or public?

File: experimenter/experimenter/experiments/tests/api/v6/test_views.py (hunk 3 above)
Comment (eliserichards) [reply experimenter/experimenter/experiments/tests/api/v6/test_views.py:248]: Absolutely can 👍

File: experimenter/experimenter/experiments/api/v6/views.py (hunk 2 above)
Comment (eliserichards) [reply experimenter/experimenter/experiments/api/v6/views.py:84]: Gah! Good catch :)

File: experimenter/experimenter/experiments/api/v6/urls.py (hunk 1 above)
Comment (eliserichards) [reply experimenter/experimenter/experiments/api/v6/urls.py:31]: `ApiView`s cannot be registered with a router in the same way that other views can -- it needs to be added as a path without the router

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
After discussing this with @eliserichards we talked about how it will be much simpler to add the FML errors into the existing feature value validation [here](https://github.com/mozilla/experimenter/blob/main/experimenter/experimenter/experiments/api/v5/serializers.py#L1387-L1414) and then worry about possibly improving the workflow later as a followup.  I think we can probably close this altogether?

**Comment by eliserichards** [comment]
Just filed https://github.com/mozilla/experimenter/issues/9606 so that we can add API docs

**Pull request closed without merging**